File: src/main/java/com/rdvdev2/timetravelmod/integration/ModMenu.java
package com.rdvdev2.timetravelmod.integration;

import com.rdvdev2.timetravelmod.impl.ModConfig;
import io.github.prospector.modmenu.api.ConfigScreenFactory;
import io.github.prospector.modmenu.api.ModMenuApi;
import me.sargunvohra.mcmods.autoconfig1u.AutoConfig;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;

@Environment(EnvType.CLIENT)
public class ModMenu implements ModMenuApi {

    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return parent -> AutoConfig.getConfigScreen(ModConfig.DisplayConfig.class, parent).get();
    }
}

File: src/main/kotlin/com/rdvdev2/timetravelmod/integration/ModMenu.kt
package com.rdvdev2.timetravelmod.integration

import com.rdvdev2.timetravelmod.impl.ModConfig
import io.github.prospector.modmenu.api.ConfigScreenFactory
import io.github.prospector.modmenu.api.ModMenuApi
import me.sargunvohra.mcmods.autoconfig1u.AutoConfig
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.minecraft.client.gui.screen.Screen

@Environment(EnvType.CLIENT)
object ModMenu : ModMenuApi {

    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> = ConfigScreenFactory { parent: Screen? ->
        AutoConfig.getConfigScreen(
            ModConfig.DisplayConfig::class.java, parent
        ).get()
    }
}